src/status_im/data_store/messages.cljs

```diff
@@ -46,22 +46,37 @@
   (some-> (data-store/get-by-id message-id)
           (clojure.core/update :user-statuses user-statuses-to-map)))
 
+(defn get-messages
+  [messages]
+  (->> messages
+       (mapv #(clojure.core/update % :user-statuses user-statuses-to-map))
+       (into '())
+       reverse
+       (keep (fn [{:keys [content-type preview] :as message}]
+               (if (command-type? content-type)
+                 (-> message
+                     (clojure.core/update :content str-to-map)
+                     (assoc :rendered-preview
+                            (when preview
+                              (generate-hiccup (read-string preview)))))
+                 message)))))
+
 (defn get-by-chat-id
   ([chat-id]
    (get-by-chat-id chat-id 0))
   ([chat-id from]
-   (->> (data-store/get-by-chat-id chat-id from c/default-number-of-messages)
-        (mapv #(clojure.core/update % :user-statuses user-statuses-to-map))
-        (into '())
-        reverse
-        (keep (fn [{:keys [content-type preview] :as message}]
-                (if (command-type? content-type)
-                  (-> message
-                      (clojure.core/update :content str-to-map)
-                      (assoc :rendered-preview
-                             (when preview
-                               (generate-hiccup (read-string preview)))))
-                  message))))))
+   (let [messages (->> (data-store/get-by-chat-id chat-id from c/default-number-of-messages)
+                       (get-messages))]
+     (if (= chat-id c/wallet-chat-id)
+       (let [wallet-messages (->> (data-store/get-by-fields {:content-type "command"}
```

Comment:
This special case for `Wallet` chat doesn't look good for me, maybe it is better to intercept incoming/outcoming messages and add them to `Wallet` chat. Imagine that we will be able to remove messages from any chat or remove chat at all. In result, transactions data will be lost for `Wallet` history.